app/Actions/Album/Extensions/LocationData.php

```diff
@@ -21,7 +21,7 @@ public function photosLocationData(Builder $photos_sql)
 
 		$return_photos = [];
 		$photo_counter = 0;
-		$photos = $photos_sql->select('album_id', 'photos.id', 'latitude', 'longitude', 'small', 'small2x', 'takestamp', 'thumb2x', 'thumbUrl', 'photos.title', 'type', 'url')
```

Comment:
The point of this was to reduce the memory footprint of Lychee :)
But nevermind. :)

Reply:
OK, I already assumed that. However, fetching the object partially from DB comes with to problems: a) We cannot use `toReturnArray` to serialize a Photo model into JSON. The current conversion routines which convert booleans into strings throw an exception on null values, if those attributes have not been fetched. b) Reporting partial object to the front-end ist difficult to handle for front-ends which don't directly use the response for displaying purpose, but feed the response into a DB for caching purposes and then drive the UI from that local shadow DB. Essentialy, the frontend would need to know, if a null value is null, because it is really null (also in the backend DB) and hence must be updated in the local shadow DB or if it is just a null value because it has not been fetched and thus has been left out of the reponse. Of course, there are solutions for that problem.

In essence, I decided to not bother with that problem right now given the fact that the PHP backend needs some major refactoring anyway. (Future outlook: Don't covert model objects into response arrays at some abitrary point in code, but only at the very end just before sending the reponse to the client.) In prospect of this refactoring it is even easier to have only one method (`toReturnArray`) instead of two which need attention. So I decided to take that efficiency penalty.

Reply:
As a rule I don't think we should be sending everything anyway unless explicitly requested, and if that hurts the memory footprint then even more so. I suspect there are more users on low memory systems than writing custom front ends! But I don't know what a back end refactor would look like, and I'd probably rather send more over the wire than have the code look a mess. I'll trust @ildyria's judgement.

Reply:
@d7415: In general, I would agree with you. However, there is so much duplication of data right now in the code, that this is the least problem. There is one approach, IHHO the _right_, approach to solve this (and other) issues. But this is a completely different story and requires much more refactoring. I will post some thoughts about that to Gitter soon. In essence, this would entirely drop this hand-crafted implementations of `toReturnArray` and use `toJSON` of the Laravel middleware. So at the moment, I would leave it as it is.